### PR TITLE
Update generated documentation

### PR DESCRIPTION
**Pull request by jason-chan-lf**
- Summary of doxygen config changes
  - `EXTRACT_STATIC = YES` so that public static classes will be documented
  - `USE_MDFILE_AS_MAINPAGE = README.md` so that the index page will contain the README.md content
  - The following settings are used so that protected members are not documented `MACRO_EXPANSION = YES`, `EXPAND_ONLY_PREDEF = YES`, `PREDEFINED = protected=private`
  - `SORT_BRIEF_DOCS = YES` so that class list is sorted
- fix inheritance images by using the `graphviz` package in the pipeline
- remove all `/// <inheritdoc/>` since it doesn't work with doxygen 1.9.1 version
- add title to changelog.md so that doxygen generates a changelog page with the correct title
- update documentation to specify Laserfiche Cloud and Laserfiche Self-Hosted.
- update `<returns>` documentation so that doxygen documentation will automatically link to the class
- other various documentation updates
- change resource file to internal 

To verify the documentation looks ok, you can download the documentation artifact linked in the pipeline

### CODE REVIEW

File: src/APIServer/TokenClientPartial.cs [hunk 1]
```diff
@@ -2,6 +2,16 @@
 
 namespace Laserfiche.Api.Client.APIServer
 {
+    /// <summary>
+    /// The Laserfiche Self-Hosted token route API client.
```
Comment (samadpaydar3) [inline src/APIServer/TokenClientPartial.cs]: I am not sure if changing this to the following makes it easier to read, as currently the the sequence is a bit long
The API client for the Laserfiche Self-Hosted token route.

File: src/ApiException.cs [hunk 2]
```diff
@@ -4,6 +4,9 @@
 
 namespace Laserfiche.Api.Client
 {
+    /// <summary>
+    /// An exception thrown from error API responses.
```
Comment (samadpaydar3) [inline src/ApiException.cs]: It reads to me like an exception is thrown from a response, while the exception is just included in the response. But I can't make a good suggestion

File: src/OAuth/AccessKey.cs [hunk 3]
```diff
@@ -11,18 +11,38 @@ namespace Laserfiche.Api.Client.OAuth
...
         public string ClientId { set; get; }
 
+        /// <summary>
+        /// The app's json web key.
+        /// </summary>
         [JsonProperty("jwk")]
         public JsonWebKey Jwk { set; get; }
 
+        /// <summary>
+        /// Create an AccessKey using the base-64 encoded access key associated with an OAuth service app,
```
Comment (samadpaydar3) [inline src/OAuth/AccessKey.cs]: Creates

File: src/ApiException.cs (hunk 2 above)
Comment (jason-chan-lf) [reply src/ApiException.cs]: I followed a similar format to dotnet Exception class description `Represents errors that occur during application execution.`
https://learn.microsoft.com/en-us/dotnet/api/system.exception?view=net-7.0

File: src/APIServer/TokenClientPartial.cs (hunk 1 above)
Comment (jason-chan-lf) [reply src/APIServer/TokenClientPartial.cs]: I updated the description. thanks

File: src/OAuth/AccessKey.cs (hunk 3 above)
Comment (jason-chan-lf) [reply src/OAuth/AccessKey.cs]: updated

**Review by samadpaydar3 — APPROVED** [review]

**Pull request merged**